sections/futures/Trade/DepositMarginModal.tsx

```diff
@@ -57,46 +51,36 @@ const DepositMarginModal: React.FC<DepositMarginModalProps> = ({
 	);
 
 	const ethPriceRate = React.useMemo(
-		() => getExchangeRatesForCurrencies(exchangeRates, Synths.sETH, selectedPriceCurrency.name),
+		() => newGetExchangeRatesForCurrencies(exchangeRates, Synths.sETH, selectedPriceCurrency.name),
 		[exchangeRates, selectedPriceCurrency.name]
 	);
 
-	const gasPrice = ethGasPriceQuery?.data?.[gasSpeed]
-		? parseGasPriceObject(ethGasPriceQuery?.data?.[gasSpeed])
-		: null;
+	const gasPrice = ethGasPriceQuery.data != null ? ethGasPriceQuery.data[gasSpeed] : null;
+
+	const depositTxn = useSynthetixTxn(
+		`FuturesMarket${market?.substring(1)}`,
+		'transferMargin',
+		[wei(!!amount ? amount : 0).toBN()],
+		gasPrice || undefined,
+		{ enabled: !!market && !!amount && !disabled }
+	);
 
 	const transactionFee = React.useMemo(
-		() => getTransactionPrice(gasPrice, gasLimit, ethPriceRate),
-		[gasPrice, gasLimit, ethPriceRate]
+		() =>
+			newGetTransactionPrice(
```

Comment:
is there an "old" `getTransactionPrice` and can we remove it?

Reply:
Yes there is, @JChiaramonte7. We can't remove it yet, since we use it in other parts of the application that are not in the scope of this PR. Once we migrate those to use the `useSynthetixTxn` hook, then we can delete the old one and rename this.